docs/dev/table/streaming/joins.md

```diff
@@ -0,0 +1,93 @@
+---
+title: "Joins"
+nav-parent_id: streaming_tableapi
+nav-pos: 3
+---
+<!--
+Licensed to the Apache Software Foundation (ASF) under one
+or more contributor license agreements.  See the NOTICE file
+distributed with this work for additional information
+regarding copyright ownership.  The ASF licenses this file
+to you under the Apache License, Version 2.0 (the
+"License"); you may not use this file except in compliance
+with the License.  You may obtain a copy of the License at
+
+  http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+
+In batch processing joins are relatively easy, since we are working on a bounded completed data sets.
+In stream processing things are a little bit more complicated, especially when it comes to the issue how to handle that data can change over time.
+Because of that, there are couple of ways to actually perform the join using either Table API or SQL.
+
+For more information regarding syntax please check Joins sections in [Table API](../tableApi.html#joins) and [SQL](../sql.html#joins).
+
+* This will be replaced by the TOC
+{:toc}
+
+Regular Joins
+-------------
+
+This is the most basic case in which any new records or changes to either side of the join input are visible and are affecting whole join result.
+If there is a new record on the left side, it will be joined with all of the previous and future records on the other side.
+
+Such semantic has an important limitation:
+it requires to keep both sides of the join input on the state indefinitely and resource usage will grow indefinitely as well.
+
+Example:
+{% highlight sql %}
+SELECT * FROM Orders
+INNER JOIN Product
+ON Orders.productId = Product.id
+{% endhighlight %}
+
+Time-windowed Joins
+-------------------
+
+In this case we are restricting scope of the join to some time window.
```

Comment:
"A time-windowed join is defined by a join predicate that checks if the time attributes(link) of the input records are within a time-window. Since time attributes are quasi-monontic increasing, Flink can remove ..."